src/Playwright/Transport/StdIOTransport.cs

```diff
@@ -155,58 +129,35 @@ private void Dispose(bool disposing)
 
         private async Task GetResponseAsync(CancellationToken token)
         {
-            try
-            {
-                var stream = _process.StandardOutput;
-                byte[] buffer = new byte[DefaultBufferSize];
+            var stream = _process.StandardOutput;
 
-                while (!token.IsCancellationRequested && !_process.HasExited)
-                {
+            while (!token.IsCancellationRequested && !_process.HasExited)
+            {
+                byte[] lengthBuffer = new byte[4];
 #if NETSTANDARD
 #pragma warning disable CA1835 // We can't use ReadOnlyMemory on netstandard
-                    int read = await stream.BaseStream.ReadAsync(buffer, 0, DefaultBufferSize, token).ConfigureAwait(false);
+                await stream.BaseStream.ReadAsync(lengthBuffer, 0, 4, token).ConfigureAwait(false);
 #pragma warning restore CA1835
 #else
-                    int read = await stream.BaseStream.ReadAsync(new(buffer, 0, DefaultBufferSize), token).ConfigureAwait(false);
+                await stream.BaseStream.ReadAsync(new(lengthBuffer, 0, 4), token).ConfigureAwait(false);
```

Comment:
This currently assumes that the buffer will be fully read which might not be the case. The code before was correct.

Check if `ReadAsync` actually read all 4 bytes if not keep reading.